go-controller/pkg/clustermanager/network_cluster_controller.go

```diff
@@ -317,7 +324,12 @@ func (ncc *networkClusterController) updateNetworkStatus(nodeName string, handle
 	if err := ncc.statusReporter(netName, "NetworkClusterController", condition, events...); err != nil {
 		return fmt.Errorf("failed to report network status: %w", err)
 	}
-	ncc.nodeErrors[nodeName] = errorMsg
+
+	if nodeHasError {
```

Comment:
I think this part is all you need to fix, but just make it
```
if errorMsg == "" {
    delete(ncc.nodeErrors, nodeName)
} else {
    ncc.nodeErrors[nodeName] = errorMsg
}
```
and then everything else should work? It was not expected that `nodeErrors` map has empty errors.

Reply:
i dont think so because lets say we process node1, it no longer has an error, so now we search for the next error node and we find one. We don't use its error message on line 315-319, we use the original error message that was for node1

Reply:
that is I think expected (maybe I should improve the function comment to make it more obvious)
so the idea is that UDN condition doesn't change until we pick a new `reportedErrorNode`, but we still report the event with the original error message for the original node (if error is not empty)

Reply:
i see i'll fix it and add a comment